tempodb/backend/block_meta.go

```diff
@@ -89,6 +90,28 @@ type CompactedBlockMeta struct {
 	CompactedTime time.Time `json:"compactedTime"`
 }
 
+func (b *CompactedBlockMeta) UnmarshalJSON(data []byte) error {
```

Comment:
I experimented with the alias here as well, but the embedding throws it off.  I think we keep the custom change without the alias here unless I missed something obvious.